public/apps/configuration/utils/role-list-utils.tsx

```diff
@@ -85,3 +87,10 @@ export function buildSearchFilterOptions(roleList: any[], attrName: string) {
     .map(e => ({ value: e }))
     .value();
 }
+
+// Submit request to delete given roles. No error handling in this function.
+export async function requestDeleteRoles(http: HttpStart, roles: string[]) {
+  roles.forEach(async r => {
+    await http.delete(`${API_ENDPOINT_ROLES}/${r}`);
+  });
```

Comment:
should we use _bluk api

https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/bulk_examples.html

you could get aggregate execution result from bulkt response for each action.

Reply:
We are using http (which of type HttpSetup). It doesn't provide a bulk api ([reference](https://github.com/elastic/kibana/blob/0d45d9b53f0bd329c642e6c60f18377f9b666842/docs/development/core/public/kibana-plugin-core-public.httpsetup.md)). And this calls security plugin server API, not elastic standard API.

Reply:
raw http request should be supported if it is low level api.

Reply:
suggestion to promise.all your await method, and add exception handling.

Reply:
Do you mean like 
~~~
await Promise.all(roles.map(r => http.delete(`${API_ENDPOINT_ROLES}/${r}`)));
~~~

Reply:
~~~
await Promise.all(roles.map(r => http.delete(`${API_ENDPOINT_ROLES}/${r}`)));
~~~
triggers 500 internal server error when deleting multiple roles. Root cause under investigation.